### PR TITLE
Assign a name to iframes when loading the initial about:blank

### LINKED ISSUE #21886: Slither.io very broken
When loading the page, at first it flashes something that looks like it may be correct rendering of start screen, then it switches to 
https://www.dropbox.com/s/w4sfvafacrsd513/Screenshot%202018-10-07%2021.59.35.png?dl=0

### PR DESCRIPTION
**Pull request by notriddle**
Before, it would assign the name too late, causing scripts (which will not wait for another tick) to accidentally spawn pop-up windows instead of loading into the iframe.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #21886
- [x] There are tests for these changes

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21976)

### CODE REVIEW

**Comment by notriddle** [comment]
# Warning: the test case in this PR doesn't pass yet

This PR allows slither.io to be opened and played.

However, due to a second bug in the iframe implementation, the `load` event handler which performs the assertion is being called too early. It should be called only when the form is finished, but is instead being called by the initial `about:blank` document.

**Comment by notriddle** [comment]
```
{"status": "TIMEOUT", "group": "default", "message": "Test timed out", "stack": null, "subtest": "Targeting nested browsing contexts", "test": "/html/browsers/windows/targeting-cross-origin-nested-browsing-contexts.html", "line": 170625, "action": "test_result", "expected": "FAIL"}
{"status": "TIMEOUT", "group": "default", "message": null, "stack": null, "subtest": null, "test": "/html/browsers/windows/targeting-cross-origin-nested-browsing-contexts.html", "line": 170626, "action": "test_result", "expected": "OK"}
```

File: tests/wpt/web-platform-tests/html/semantics/forms/form-submission-target/form-target-iframe.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<title>Form targetted at iframe</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+async_test(function(t) {
+  window.addEventListener("load", function() {
```
Comment (jdm) [inline tests/wpt/web-platform-tests/html/semantics/forms/form-submission-target/form-target-iframe.html]: This should pass `t.step_func(function() {` instead.

File: tests/wpt/web-platform-tests/html/semantics/forms/form-submission-target/form-target-iframe.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    var form = document.createElement("form");
+    form.target = "frame";
+    form.action = "form-target-iframe-helper.py";
+    form.method = "POST";
+    var input = document.createElement("input");
+    input.name = "n";
+    form.appendChild(input);
+    document.documentElement.appendChild(form);
+    form.submit();
+    frame.addEventListener("load", function() {
```
Comment (jdm) [inline tests/wpt/web-platform-tests/html/semantics/forms/form-submission-target/form-target-iframe.html]: This should also pass `t.step_func(function() {`

File: tests/wpt/web-platform-tests/html/semantics/forms/form-submission-target/form-target-iframe.html [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      if (frame.contentWindow.location.href != "about:blank") {
```
Comment (jdm) [inline tests/wpt/web-platform-tests/html/semantics/forms/form-submission-target/form-target-iframe.html]: Could we make this check for the actual expected value instead? Maybe using `new URL("form-target-iframe-helper.py", window.location)`?

**Review by jdm — CHANGES_REQUESTED** [review]
Thanks for looking into this so carefully!

**Comment by notriddle** [comment]
@bors-servo retry

Out-of-memory

**Pull request merged**